Assets/TopicsTestTokenVendingMachine.cs

```diff
@@ -0,0 +1,269 @@
+                                        textAreaString += "Error receiving message, cancelling...";
+                                        cts.Cancel();
+                                        break;
+                                }
+                            }
+                        }
+                        finally
+                        {
+                            Debug.Log("Disposing subscription to topic " + TopicName);
+                            subscription.Dispose();
+                        }
+
+                        break;
+                    case TopicSubscribeResponse.Error error:
+                        Debug.LogError(String.Format("Error subscribing to a topic: {0}", error.Message));
+                        textAreaString += "Error trying to connect to chat, cancelling...";
+                        cts.Cancel();
+                        break;
+                }
+            });
+        }
+        finally
+        {
+            Debug.Log("Disposing cache and topic clients...");
+            client.Dispose();
+            topicClient.Dispose();
+        }
+    }
+
+    public void PublishMessage()
+    {
+        // Unlike the other examples, because we're using the Token Vending Machine,
+        // we now no longer need to send the user's name in the message since all
+        // subscribers will receive the publishers' names via the message's TokenId.
+        string message = inputTextField.text;
+        Task.Run(async () =>
+        {
+            Debug.Log("About to publish message: " + message);
+            if (cts != null && !cts.IsCancellationRequested)
+            {
+                var publishResponse =
+                    await topicClient.PublishAsync(cacheName, TopicName, message);
+                switch (publishResponse)
+                {
+                    case TopicPublishResponse.Success:
+                        Debug.Log("Successfully published message " + message);
+                        break;
+                    case TopicPublishResponse.Error error:
+                        Debug.LogError(String.Format("Error publishing a message to the topic: {0}", error.Message));
+                        cts.Cancel();
+                        break;
+                }
+            }
+            else
+            {
+                Debug.LogError("Could not publish message since cancellation already occurred");
+            }
+        });
+        inputTextField.text = "";
+        inputTextField.ActivateInputField();
+    }
+
+    private IEnumerator GetTokenFromVendingMachine(string name)
+    {
+        textAreaString = "LOADING...";
+        loading = true;
+
+        // Set your Token Vending Machine URL here. For more information, see:
+        // https://github.com/momentohq/client-sdk-javascript/tree/main/examples/nodejs/token-vending-machine
+        string uri = "https://9jkmukxn68.execute-api.us-west-2.amazonaws.com/prod?name=" + UnityWebRequest.EscapeURL(name);
```

Comment:
Can we do something here that will cause a fast and obvious failure if they haven't updated it? Like, set it to null by default and then in the code, check to see if it is null and throw an error?

it's fine for you to keep this URL in here as an example, I just want to make sure that the failure they get if they run it without updating this is really obvious as to what needs to be fixed.

Reply:
Definitely, good idea. It's updated now so that if you just run the example, you'll see this:
![image](https://github.com/momentohq/momento-unity-demo/assets/276325/d6936110-e177-476d-a322-94db812f1b87)

If you do specify a URL but it's not valid, then after typing in a name, you'll see:
![image](https://github.com/momentohq/momento-unity-demo/assets/276325/10b3df33-92f3-438f-b3ca-17c96f074dc7)